### PR TITLE
Webpack update

### PR DESCRIPTION
**Pull request by jkozol**
The webpack config has been simplified and modernized. Unneeded imports
are cleared from it. The folder public/ has now been refactored to dist/
and the dist build process is simpler. ejs is no longer used in favor of
an index.html file. Our babel config is modernized and moved to a
.babelrc.json.

The routing is now handled by react-router. This simplifies the routing
greatly and gives us quicker flexibility to modify routing and props
passing.

The translations workflow is improved to work on a translations/ dir
with translations imported from weblate directly. The npm scripts are
also simplified to reduce usage of unnneccesray library usage to convert
pot files.

This also removes the need for cockpit-composer-weblate. See https://github.com/osbuild/cockpit-composer/pull/1562 for example of how translations will be updated.

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -1,105 +1,48 @@
...
-const CopyWebpackPlugin = require("copy-webpack-plugin");
-const HtmlWebpackPlugin = require("html-webpack-plugin");
-const CleanWebpackPlugin = require("clean-webpack-plugin");
 const MiniCssExtractPlugin = require("mini-css-extract-plugin");
-const babelConfig = require("./babel.config");
 
 // absolute path disables recursive module resolution, so build a relative one
-const nodedir = path.relative(process.cwd(), path.resolve(process.env.SRCDIR || __dirname, "node_modules"));
-
-const [mode, devtool] =
```
Comment (jkozol) [inline webpack.config.js:12]: This removal is tracked here: https://github.com/osbuild/cockpit-composer/issues/1576

File: Makefile [hunk 2]
```diff
@@ -24,8 +24,6 @@ translations: $(WEBPACK_TEST)
 install: all
 	mkdir -p /usr/share/cockpit/composer
 	cp -r public/* /usr/share/cockpit/composer
-	mkdir -p /usr/share/metainfo/
-	cp io.weldr.cockpit-composer.metainfo.xml /usr/share/metainfo/
```
Comment (Gundersanne) [inline Makefile:28]: Isn't this required? How is cockpit-composer exposed now?

File: cockpit-composer.spec.in [hunk 3]
```diff
@@ -30,15 +30,11 @@ the cloud. It integrates into Cockpit as a frontend for osbuild.
 
 %install
 mkdir -p %{buildroot}/%{_datadir}/cockpit/composer
-cp -a public/dist/* %{buildroot}/%{_datadir}/cockpit/composer
-mkdir -p %{buildroot}/%{_datadir}/metainfo/
-appstream-util validate-relax --nonet io.weldr.cockpit-composer.metainfo.xml
-cp -a io.weldr.cockpit-composer.metainfo.xml %{buildroot}/%{_datadir}/metainfo/
```
Comment (Gundersanne) [inline cockpit-composer.spec.in:36]: Same here, how is cockpit-composer exposed now? Could use some more detail in the commit message.

File: .github/workflows/check-translations.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+          npm i
+          npm run translations:extract
+          if [ -n "$(git status --porcelain)" ]; then
+            echo
+            echo "translations/en.json needs to be generated"
+            echo "Please run `npm run translations:extract`"
+            exit "1"
+          else
+            exit "0"
+          fi
```
Comment (Gundersanne) [inline .github/workflows/check-translations.yml:24]: Personal nitpick, trailing new lines :)

**Review by Gundersanne — COMMENTED** [review]
I think some of the commits could use a bit more info the in the commit message itself.

File: Makefile (hunk 2 above)
Comment (jkozol) [reply Makefile:28]: Under `dist` there is a `manifest.json` file which declares cockpit-composers placement on the dashboard. This metainfo file did not seem to be doing anything. It was attached to the applications section of cockpit but, at least locally, did not have any effect.

File: cockpit-composer.spec.in (hunk 3 above)
Comment (jkozol) [reply cockpit-composer.spec.in:36]: See above

File: .github/workflows/check-translations.yml (hunk 4 above)
Comment (jkozol) [reply .github/workflows/check-translations.yml]: Fixed :)

File: Makefile (hunk 2 above)
Comment (Gundersanne) [reply Makefile:28]: It's used in the applications tool in cockpit. I'd definitely check with them before removing it.

Without it Image Builder won't actually show up under /apps.

Comment (jkozol) [reply Makefile:28]: I will reach out to them. Does installing image-builder from cockpit work for you? It's broken for me. I created this tracker #1578. To reimplement the applicaitons list for cockpit composer.

Comment (Gundersanne) [reply Makefile:28]: I think you can only ever remove an application from the /apps page.

Comment (jkozol) [reply Makefile:28]: Ok. I readded it in but not in the makefile since the install script has been removed. It is in the specfile.

**Review by lucasgarfield — APPROVED** [review]
Great work!

**Pull request merged**